File: src/app/Aluno.php
<?php

namespace App;

class Aluno
{
    private int $id;
    private string $nome;

    /** @var float[] */
    private array $notasEscolares = [];

    public function __construct(int $id, string $nome)
    {
        $this->id = $id;
        $this->nome = $nome;
    }

    public function getId(): int
    {
        return $this->id;
    }

    public function getNome(): string
    {
        return $this->nome;
    }

    /** @return float[] */
    public function getNotasEscolares(): array
    {
        return $this->notasEscolares;
    }

    public function addNotaEscolar(float $nota): void
    {
        $this->notasEscolares[] = $nota;
    }

    public function calcularMediaEscolar(): float
    {
        $notas = $this->getNotasEscolares();
        if (count($notas) === 0) {
            return 0.0;
        }
        $soma = array_sum($notas);
        return $soma / count($notas);
    }

}

File: 1-modelagem-codigo-python/app/aluno.py
class Aluno:
    def __init__(self, id: int, nome: str):
        self.__id = id
        self.__nome = nome
        self.__notas_escolares: list[float] = []

    def get_id(self) -> int:
        return self.__id

    def get_nome(self) -> str:
        return self.__nome

    def get_notas_escolares(self) -> list[float]:
        return self.__notas_escolares

    def add_nota_escolar(self, nota: float) -> None:
        self.__notas_escolares.append(nota)

    def calcular_media_escolar(self) -> float:
        if not self.__notas_escolares:
            return 0.0
        return sum(self.__notas_escolares) / len(self.__notas_escolares)